content/lessons/chapter-8/building-blocks-4.tsx

```diff
@@ -1,86 +1,127 @@
 'use client'
 
-import { useProceed, useTranslations } from 'hooks'
-import { ChapterIntro, CodeExample, HolocatQuestion } from 'ui'
-
-import { Button } from 'shared'
-import { useEffect, useState } from 'react'
-import { getData } from 'api/data'
-import { Data } from 'types'
-import { chapters } from 'content/chapters'
+import { ScriptingChallenge, LessonInfo, CodeExample } from 'ui'
+import { EditorConfig } from 'types'
+import { useTranslations } from 'hooks'
+import { Text } from 'ui'
+import { useState } from 'react'
+import { getLessonKey } from 'lib/progress'
+import { useDataContext } from 'contexts/DataContext'
+import { getLanguageString } from 'lib/SavedCode'
+import { useAtom } from 'jotai'
+import { accountAtom } from 'state/state'
 
 export const metadata = {
-  title: 'chapter_four.address_one.title',
+  title: 'chapter_eight.building_blocks_four.nav_title',
+  navigation_title: 'chapter_eight.building_blocks_four.nav_title',
   key: 'CH8BBK4',
 }
 
 export default function BuildingBlocks4({ lang }) {
-  const proceed = useProceed()
   const t = useTranslations(lang)
+  const [account] = useAtom(accountAtom)
+  const { currentLanguage } = useDataContext()
+  const [privateKey, setPrivateKey] = useState('')
+
+  if (account && !privateKey) {
+    setPrivateKey(account?.private_key.toString())
+  }
+
+  const javascript = {
+    program: `
+console.log(getBlockHeight())
+console.log("KILL")`,
+    defaultFunction: {
+      name: 'privateKeyToPublicKey',
+      args: ['privateKey'],
+    },
+    defaultCode: `const Bitcoin = require('@0tuedon/bitcoin_rpcjs')
+const CODE_CHALLENGE_2_HEIGHT = 6929996;
+const hashes = Bitcoin.rpc('getblocksbyheight', CODE_CHALLENGE_2_HEIGHT);
```

Comment:
I think the user should try to create this temselves, we might want to add some additional copy to help them along though
``` suggestion
```